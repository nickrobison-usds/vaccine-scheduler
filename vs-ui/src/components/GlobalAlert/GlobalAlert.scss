.vs-alert {
  position: absolute;
  top: 0;
  right: 0;
  display: flex;
  justify-content: space-around;
  margin-top: 1vh;
  margin-right: 1vh;
  width: 40vh;
  z-index: 10;
}
