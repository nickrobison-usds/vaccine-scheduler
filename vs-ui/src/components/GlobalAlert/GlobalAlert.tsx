import { Alert } from "@trussworks/react-uswds";
import React from "react";
import "./GlobalAlert.scss";

export type AlertType = "success" | "warning" | "error" | "info";

export interface AlertProps {
  typ: AlertType;
  text: string;
  icon: boolean;
}

export const GlobalAlert: React.FC<AlertProps> = (props) => {
  return (
    <Alert type={props.typ} noIcon={!props.icon} className="vs-alert">
      {props.text}
    </Alert>
  );
};
