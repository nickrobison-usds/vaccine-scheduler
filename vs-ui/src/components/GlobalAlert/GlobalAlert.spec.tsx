import { render } from "@testing-library/react";

import { GlobalAlert } from "./GlobalAlert";

describe("GlobalAlert", () => {
  it("renders with icon", () => {
    const { container } = render(
      <GlobalAlert typ="warning" text="icon" icon={true} />
    );
    expect(container).toMatchSnapshot();
  });
  it("renders without an icon", () => {
    const { container } = render(
      <GlobalAlert typ="warning" text="no icon" icon={false} />
    );
    expect(container).toMatchSnapshot();
  });
});
