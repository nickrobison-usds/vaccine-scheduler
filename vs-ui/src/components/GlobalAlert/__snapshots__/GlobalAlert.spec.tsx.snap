// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalAlert renders with icon 1`] = `
<div>
  <div
    class="usa-alert usa-alert--warning vs-alert"
    data-testid="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        icon
      </p>
    </div>
  </div>
</div>
`;

exports[`GlobalAlert renders without an icon 1`] = `
<div>
  <div
    class="usa-alert usa-alert--warning usa-alert--no-icon vs-alert"
    data-testid="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        no icon
      </p>
    </div>
  </div>
</div>
`;
